internal/budgetadjustments/flags.go

```diff
@@ -0,0 +1,96 @@
+package budgetadjustments
+
+import (
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/nobl9/sloctl/internal/csv"
+)
+
+const (
+	flagAdjustment = "adjustment-name"
+	flagFrom       = "from"
+	flagTo         = "to"
+	flagProject    = "project"
+	flagSloName    = "slo-name"
+)
+
+func mustRegisterOutputFormatFlags(
+	cmd *cobra.Command,
+	outputFormat, fieldSeparator, recordSeparator *string,
+) {
+	cmd.PersistentFlags().StringVarP(outputFormat, "output", "o", "yaml",
+		`Output format: one of yaml|json|csv.`)
+
+	cmd.PersistentFlags().StringVarP(fieldSeparator, csv.FieldSeparatorFlag, "",
+		csv.DefaultFieldSeparator, "Field Separator for CSV.")
+
+	cmd.PersistentFlags().StringVarP(recordSeparator, csv.RecordSeparatorFlag, "",
+		csv.DefaultRecordSeparator, "Record Separator for CSV.")
+
+	if err := cmd.PersistentFlags().MarkHidden(csv.RecordSeparatorFlag); err != nil {
+		panic(err)
+	}
+}
+
+func mustRegisterAdjustmentFlag(cmd *cobra.Command, storeIn *string) {
+	cmd.Flags().StringVar(storeIn, flagAdjustment, "", "Name of the Adjustment.")
+	if err := cmd.MarkFlagRequired(flagAdjustment); err != nil {
+		panic(err)
+	}
+}
+
+func registerProjectFlag(cmd *cobra.Command, storeIn *string) {
+	cmd.Flags().StringVarP(storeIn, flagProject, "", "",
+		"Name of the project. Required when sloName is defined.")
+}
+
+func registerSloNameFlag(cmd *cobra.Command, storeIn *string) {
+	cmd.Flags().StringVarP(storeIn, flagSloName, "", "",
+		"Name of the SLO. Required when sloName is defined.")
+}
+
+type TimeValue struct{ time.Time }
+
+const (
+	timeLayout       = time.RFC3339
+	timeLayoutString = "RFC3339"
+)
+
+func (t *TimeValue) String() string {
+	if t.IsZero() {
+		return ""
+	}
+	return t.Format(timeLayout)
+}
+
+func (t *TimeValue) Set(s string) (err error) {
+	t.Time, err = time.Parse(timeLayout, s)
```

Comment:
You should catch error and throw own like 'Date in invalid format' because now it's golang's error like ''parsing time "budget-adjustment-recurr-2" as "2006-01-02T15:04:05Z07:00": cannot parse "budget-adjustment-recurr-2" as "2006" I know what's going on but this golang date format date can be unreadable for users.

Reply:
done, changed to:
```
Error: invalid argument "2023-09-23T20:46:00Za" for "--to" flag: to does not match 2006-01-02T15:04:05Z07:00 format
```